# HG changeset patch
# User BelovaMM
# Parent  4f336738d83ab3e577ae4f1b4d81f565b2e1baed
refs #DEVBO-753 Проверка возраста MAX ID. Отладка. #BTS-2584 #BTS-2585
BTS-2584 Необходимо не записывать в ППО Trade House (не отображать в отчёте по параметрам АРМ кассира) значения ряда новых конфигурационных параметров, переданных с кассы.
BTS-2585 ТН не отправляет значение логина прокси сервера на кассу

https://msk-exp-jira.expertek.local/browse/DEVBO-753
https://msk-exp-jira.expertek.local/browse/BTS-2584
https://msk-exp-jira.expertek.local/browse/BTS-2585

diff --git a/gbl/gis.w b/gbl/gis.w
--- a/gbl/gis.w
+++ b/gbl/gis.w
@@ -1014,6 +1014,11 @@ ASSIGN FRAME {&FRAME-NAME}
               temp-thbj-attr.property-value-character = gisAdress.
            when {&attr-gisMT_proxyLogin} 
            then do:
+               if temp-thbj-attr.property-value-character <> login
+              then do: 
+                 create thbjattr-list.
+                 buffer-copy temp-thbj-attr to thbjattr-list.
+              end.  
               temp-thbj-attr.property-value-character = login.
            end.   
            when  {&attr-gisMT_proxyPswd} 
diff --git a/str/send-all-work-cashp1.p b/str/send-all-work-cashp1.p
--- a/str/send-all-work-cashp1.p
+++ b/str/send-all-work-cashp1.p
@@ -33,6 +33,9 @@ define variable Mfirst as logical no-und
 function getValueConvert returns character  (igroup as char,
                                              iparam as char,
                                              ivalue as char):
+   define variable v-param-empty as character no-undo.
+   v-param-empty = "MACC_IP,MaxApiToken,Proxy_*,LmCHzLogin,LmCHzPass,TspiotHostAddr".
+
    if igroup eq "FUCO"
    then do:
       if iparam eq "TrkDispAdr"
@@ -41,7 +44,7 @@ function getValueConvert returns charact
    end.
    else if igroup eq "GS1"
    then do:
-      if iparam eq "MACC_IP"
+      if can-do(v-param-empty, iparam)
       then 
          ivalue = "".
    end.
